src/infrastructure/content/ContentCommunicator.java

```diff
@@ -1,24 +1,256 @@
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+			return;
+		}
+		serverIPAddress += ":";
+		serverIPAddress += String.valueOf(port);
+		logMessage = "content: server's ipAddress in ContentCommunicator class created";
+		logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
+		imageMap.put(userName, userImage);
+		jsonObject.put("meta", "newUser");
+		jsonObject.remove("ipAddress");
+		communicator.send(serverIPAddress, jsonObject.toString(), "contentServer");
+		logMessage = "content: desired message passed to networking to send it to server";
+		logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
+		logMessage = "content: Successfully finished the initialiseUser method of ContentCommunicator class";
+		logger.log(ModuleID.INFRASTRUCTURE, LogLevel.SUCCESS, logMessage);
+	}
 	
+	/**
+	 * This method gets the message from UI module and then send it via networking module.
+	 * A meta field is added to it indicating that it is message, before sending to server.
+	 * An username field is added indicating the owner of the actual message.
+	 * @param message - This is a JSON string which contains a JSON object with key "message"
+	 */
 	@Override
 	public void sendMessageToContent(String message) {
-    	
-    }
+		JSONObject jsonObject;
+		try {
+			jsonObject = new JSONObject(message);
+		} catch(Exception e) {
+			StringWriter sw = new StringWriter();
+			e.printStackTrace(new PrintWriter(sw));
+			logMessage = "content: "+sw.toString();
+			logger.log(ModuleID.INFRASTRUCTURE, LogLevel.ERROR, logMessage);
+			return;
+		}
+		jsonObject.put("meta", "message");
+		jsonObject.put("username",userName);
+		communicator.send(serverIPAddress, jsonObject.toString(), "contentServer");
+		logMessage = "content: desired message passed to networking to send it to server";
+		logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
+		logMessage = "content: Successfully finished the sendMessageToContent method of ContentCommunicator class";
+		logger.log(ModuleID.INFRASTRUCTURE, LogLevel.SUCCESS, logMessage);
+	}
 	
+	/**
+	 * This method clears all the maps created by content module locally and informs the
+	 * server that the client is leaving the board by sending a JSON String which contains
+	 * a JSON object with two keys, meta and username via networking module.
+	 */
 	@Override
 	public void notifyUserExit() {
-    	
-    }
-    
+		logMessage = "content: notifyUserExit method of ContentCommunicator is called";
+		logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
+		imageMap.clear();
+		handlerMap.clear();
+		logMessage = "content: imageMap and handlerMap of ContentCommunicator class are cleared";
+		logger.log(ModuleID.INFRASTRUCTURE, LogLevel.INFO, logMessage);
+		JSONObject jsonObject = new JSONObject();
+		jsonObject.put("meta", "userExit");
+		jsonObject.put("username", userName);
+		communicator.send(serverIPAddress, jsonObject.toString(), "contentServer");

```

Comment:
identifier is content. Change it. This identifier is used only at the client side. I've already told this